docs/0001-teleport-job-worker.md

```diff
@@ -0,0 +1,136 @@
+---
+authors: Brodie Kurczynski (brodie@kurczynski.org)
+state: draft
+---
+
+# RFD 1 - Teleport Job Worker
+
+## What
+
+A service that runs commands submitted by multiple users. These commands may run concurrently and will provide the users
+with the ability to limit resource (i.e. CPU, memory, and disk IO) usage. The service will consist of three pieces:
+
+* A reusable library implementing the functionality for working with jobs
+* An API server that wraps the functionality of the library
+* A command line interface (CLI) that communicates with the API server
+
+### Required Approvers
+
+* Engineering: (@sclevine && @gavinfrazar) || (@sclevine && @bernardjkim) || (@sclevine && @jimbishopp) || (
+  @gavinfrazar && @bernardjkim) || (@gavinfrazar @jimbishopp) || (@bernardjkim @jimbishopp)
+
+### Security
+
+Requests made to the service will use mTLS for authentication. Specifically:
+
+* TLS using the modern version 1.3
+* The cipher suite will need no configuration, it will be handled by `crypto/tls` (this was updated
+  in [Go 1.17](https://go.dev/doc/go1.17#crypto/tls))
+* X.509 public/private key pairs will be generated using RSA-4096
+
+For authorization, the service will use simple tokens to ensure that jobs can only be accessed and modified by the user
```

Comment:
How are these tokens created?

Reply:
Also, what if another user steals this token? It sounds like the system will reject them anyway since they are not the user who created the job right? Then why have a token at all?

Reply:
The token seems unnecessary even for this toy project though - users can only access a job that they created right? And we can verify their identity with mTLS.

Reply:
Ahhhh okay, that makes more sense to use the existing mTLS. The part that confused me in the challenge doc was `Use mTLS authentication and verify client certificate` followed by `Use a simple authorization scheme`. I took that as "use mTLS and then do something else for authorization".

I'll update the doc to reflect that.

Reply:
mTLS only establishes identity, i.e. authentication, (as does a secret token), so the authorization scheme is really independent from both of these things.

And I think just enforcing that only a job creator can interact with the job is fine as a simple authorization scheme for the challenge